### ISSUE TITLE
Problem with PayPal Rest + Symfony4

### ISSUE THREAD

**Issue by Loctarogar** [Payum/Payum#782]
I use Symfony4 and have an error in function doneAction
        $payment->execute($status = new GetHumanStatus($token));  <--- here
Error :
"Request GetHumanStatus{model: ArrayObject} is not supported. Make sure the gateway supports the requests and there is an action which supports this request (The method returns true). There may be a bug, so look for a related issue on the issue tracker."
My controller:
<?php declare(strict_types=1);

namespace App\Controller\API;

use FOS\RestBundle\Controller\FOSRestController;
use FOS\RestBundle\View\View;
use Symfony\Component\HttpFoundation\Response;
use Payum\Core\Request\GetHumanStatus;
use Symfony\Component\HttpFoundation\Request;
use App\Entity\Order;
use App\Entity\User;
use App\Services\CartService;
use App\Entity\Payment;
use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
use FOS\RestBundle\Controller\Annotations as Rest;
use Payum\Core\Payum;

class PaymentController extends FOSRestController
{
    /**
     * @return Payum
     */
    protected function getPayum()
    {
        return $this->get('payum');
    }

    /**
     * @param Order       $order
     * @param CartService $cartService
     * @return mixed
     * @Rest\Post("/prepare-action/{order}", name="prepare-action")
     */
    public function prepareAction(Order $order, CartService $cartService): View
    {
        if (true === $order->getIsPaid()) {
            throw new BadRequestHttpException('Order already payed.');
        }
        /* @var $user User */
        $user = $this->getUser();
        $gatewayName = 'paypal_rest';
        $storage = $this->getPayum()
            ->getStorage(Payment::class);
        /* @var $payment Payment */
        $payment = $storage->create();
        $total = $cartService->getTotalSum();
        $payment->setUserOrder($order);
        $payment->setCurrencyCode('PLN');
        $payment->setTotalAmount($total);
        $payment->setDescription('Shop payment.');
        $payment->setClientId(md5((string) $user->getId()));
        $payment->setClientEmail($user->getEmail());
        $details = [
            'invoiceDisabled' => true, // use this only when you want to hide invoice checkbox on payment form
        ];
        if (null !== $userInfo = $user->getUsername()) {
            $details = [
                'firstName' => $user->getUsername(),
            ];
        }
        $payment->setDetails($details);
        $storage->update($payment);
        $captureToken = $this->getPayum()
            ->getTokenFactory()
            ->createCaptureToken(
                $gatewayName,
                $payment,
                'payum_capture_do');

        return View::create($captureToken->getTargetUrl(), Response::HTTP_OK);
    }

    /**
     * @param Request $request
     * @return mixed
     * @throws \Exception
     * @Rest\Get("done", name="payum_capture_do")
     */
    public function doneAction(Request $request)
    {
        $token = $this->getPayum()->getHttpRequestVerifier()->verify($request);
        $payment = $this->getPayum()->getGateway($token->getGatewayName());
        $payment->execute($status = new GetHumanStatus($token));
        $paymentId = $status->getToken()->getDetails()->getId();
        $paymentClass = $status->getToken()->getDetails()->getClass();
        $storage = $this->getPayum()->getStorage($paymentClass);

        /* @var $payment Payment */
        $payment = $storage->find($paymentId);
        if (!$payment) {
            throw new NotFoundHttpException('Payment not found');
        }
        $order = $payment->getUserOrder();
        if ($status->isCaptured()) {
            // payment succeeded
            $order->setIsPaid(true);
            $order->setStatus(Order::STATUS_ACCEPTED);
        } elseif ($status->isPending()) {
            // payment pending
        } else {
            // payment failed
            $order->setIsPaid(false);
            $order->setStatus(Order::STATUS_CANCELLED);
        }
        $em = $this->getDoctrine()->getManager();
        $em->persist($order);
        $em->flush();

        return View::create($payment, Response::HTTP_OK);
    }
}

My payum.yaml:

payum:
    security:
        token_storage:
            App\Entity\PaymentToken:
                filesystem:
                    storage_dir: "%kernel.root_dir%/Resources/payments"
                    id_property: hash

    storages:
        App\Entity\Payment:
            filesystem:
                storage_dir: "%kernel.root_dir%/Resources/payments"
                id_property: id

    gateways:
        offline:
            factory: offline

        paypal_rest:
            factory:  "paypal_rest"
            client_id: "someId"
            client_secret: "someSecret"
            config_path: '%kernel.root_dir%/sdk_config.ini'

**Comment by RealBC** [comment]
May i ask you if you found any solution cause i still got the error " Request GetStatus{model: ArrayObject} is not supported...."

**Comment by Loctarogar** [comment]
@RealBC  no, i still haven't any solution. I would suggest to use PayPal sandbox https://developer.paypal.com/docs/classic/lifecycle/ug_sandbox/?mark=sandbox

**Comment by RealBC** [comment]
> @RealBC no, i still haven't any solution. I would suggest to use PayPal sandbox https://developer.paypal.com/docs/classic/lifecycle/ug_sandbox/?mark=sandbox +
> https://github.com/paypal/PayPal-PHP-SDK

**Comment by ndoulgeridis** [comment]
Any progress on this one?

**Comment by Loctarogar** [comment]
@ndoulgeridis  nope

**Comment by nebia17** [comment]
YES i spent some time on it  :) I succesfully integrated it to Sylius 1.2.6 and with Symfony 3 + PayumBundle using doctrine storage (and not filesystem ...filesystem method gave me some helps to debug the famous ArrayObject error :) ) I will signed my code and try to make a repository for the next week for the first time on GitHub free of charge and use !

**Comment by RealBC** [comment]
YES i spent some time on it :) I succesfully integrated it to Sylius 1.2.6 and with Symfony 3 + PayumBundle using doctrine storage (and not filesystem ...filesystem method gave me some helps to debug the famous ArrayObject error :) ) I will sign my code and try to make a repository for the next week for the first time on GitHub free of charge and use ! PS account change sorry

**Comment by RealBC** [comment]
have a look ;)
https://github.com/RealBC/Symfony3_PaypalREST_DoctrineStorage
